### PR TITLE
feat(onboarding image): [MAG-3050] Create drop-down component

### PR DESCRIPTION
**Pull request by eTheMan80**
# <MAG-3050 / Create drop-down component>

## TL;DR

-   Create language dropdown as an extension of dropdown component
-  
-  
-  

## Details

Checklist / Reminders:

-   [x] Documentation / Story is created / adapted
-   [x] Unit tests
-   [x] CSS changes are BEM-compliant

Screenshot 2020-11-18 at 13 18 54

### CODE REVIEW

**Review by tricaalexandru — APPROVED** [review]

File: src/components/Molecules/Dropdown/Dropdown.styled.ts [hunk 1]
```diff
@@ -28,23 +28,32 @@ export const StyledContainer = styled.div`
...
+    position: relative;
+    display: inline-block;
+    margin-top: ${hasLabel ? theme.font.lineHeight.sm : 0};
+    height: ${theme.font.lineHeight.sm};
+    line-height: ${theme.font.lineHeight.sm};
+    box-shadow: ${boxShadow
+      ? theme.dimension.elevationLevel2
+      : active
+      ? theme.dimension.elevationLevel4
+      : 'none'};
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/Dropdown/Dropdown.styled.ts:42]: @matthill8286 in Craig's design https://projects.invisionapp.com/share/EPYY0ASF7WM#/screens/436204061, the active dropdown component has a thicker box-shadow than the normal. How is it best to achieve this?

**Comment by tricaalexandru** [comment]
@matthill8286 yes. your comments are making a lot of stuff more understandable!

File: src/components/Molecules/Dropdown/Dropdown.styled.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthill8286) [reply src/components/Molecules/Dropdown/Dropdown.styled.ts:42]: 😢  Designers!!! Thats all I can say about that. Not a fan but it is what it is, i'll resolve this as long as it's applied to the whole dropdown, including the dropdown list.  @eTheMan80

File: src/components/Molecules/LanguageDropDown/LanguageDropDown.interface.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { DropdownProps } from '@/components/Molecules/Dropdown/Dropdown.interface'
+
+export interface LanguageDropDownProps extends DropdownProps {
+  boxShadow?: boolean
+  curved?: boolean
+}
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/LanguageDropDown/LanguageDropDown.interface.ts]: @matthill8286 this has been updated

File: src/components/Molecules/Dropdown/Dropdown.styled.ts [hunk 3]
```diff
@@ -69,7 +79,7 @@ export const StyledList = styled.ol<ListProps>`
     z-index: 3;
     list-style: none;
     padding: ${theme.spacing.base.xs} 0;
-    margin: 0;
+    margin: 20px 0;
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/Dropdown/Dropdown.styled.ts]: @matthill8286 this has been updated

File: src/components/Molecules/LanguageDropDown/LanguageDropDown.story.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  { id: '3', label: 'Italian' },
+  { id: '4', label: 'Spanish' },
+]
+
+const commonKnobs = {
+  onChange: action('on-change'),
+  options,
+}
+
+const story = storiesOf('Molecules|Language dropdown', module)
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/LanguageDropDown/LanguageDropDown.story.tsx]: @matthill8286 this has been updated

File: src/components/Molecules/LanguageDropDown/LanguageDropDown.story.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+story.add(
+  'Default',
+  () => {
+    const knobs: DropdownProps = {
+      ...commonKnobs,
+      label: text('Label', 'type a label'),
+      initialIndex: number('initialIndex', 0),
+    }
+
+    return <LanguageDropDown {...knobs} />
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/LanguageDropDown/LanguageDropDown.story.tsx]: @matthill8286 this has been updated

File: src/components/Molecules/LanguageDropDown/LanguageDropDown.story.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import { action } from '@storybook/addon-actions'
+import { number, text } from '@storybook/addon-knobs'
+import { storiesOf } from '@storybook/react'
+import React from 'react'
+import { DropdownProps } from '@/components/Molecules/Dropdown/Dropdown.interface'
+import readme from '@/components/Molecules/Dropdown/Dropdown.readme.md'
+import { LanguageDropDown } from './LanguageDropDown'
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/LanguageDropDown/LanguageDropDown.story.tsx]: @matthill8286 this has been updated

File: src/components/Molecules/LanguageDropDown/LanguageDropDown.test.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import { mountWithTheme, renderWithTheme } from '@/testRenderer'
+import { LanguageDropDown } from './LanguageDropDown'
+
+describe('Language dropdown', () => {
+  it('renders dropdown', () => {
+    const wrapper = renderWithTheme(
+      <LanguageDropDown active={false} options={[{ label: 'one' }, { label: 'two' }]} />
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/LanguageDropDown/LanguageDropDown.test.tsx]: @matthill8286 this has been updated

File: src/components/Molecules/LanguageDropDown/LanguageDropDown.test.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    )
+    expect(wrapper).toMatchSnapshot()
+  })
+  it('renders dropdown', () => {
+    const wrapper = mountWithTheme(
+      <LanguageDropDown active={false} options={[{ label: 'one' }, { label: 'two' }]} />
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/LanguageDropDown/LanguageDropDown.test.tsx]: @matthill8286 this has been updated

File: src/components/Molecules/LanguageDropDown/LanguageDropDown.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { Dropdown } from '@/components/Molecules/Dropdown'
+import { LanguageDropDownProps } from './LanguageDropDown.interface'
+
+export const LanguageDropDown: React.FC<LanguageDropDownProps> = ({
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/LanguageDropDown/LanguageDropDown.tsx]: @matthill8286 this has been updated

File: src/components/Molecules/LanguageDropDown/LanguageDropDown.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+}) => {
+  return (
+    <Dropdown
+      options={options}
+      initialIndex={initialIndex}
+      onChange={onChange}
+      autoComplete={autoComplete}
+      boxShadow={boxShadow}
+      curved={curved}
+      label={''}
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/LanguageDropDown/LanguageDropDown.tsx]: @matthill8286 this has been updated

File: src/components/Molecules/Dropdown/Dropdown.styled.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/Dropdown/Dropdown.styled.ts]: @matthill8286 this has been updated

**Comment by matthill8286** [comment]
@eTheMan80 great work. Merged

**Pull request merged**